compiler/ast/ast_types.nim

```diff
@@ -1256,7 +1256,7 @@ type
     adSemDeprecatedCompilerOptArg   # warning promoted to error
 
   PAstDiag* = ref TAstDiag
-  TAstDiag* = object
+  TAstDiag* {.acyclic.} = object
```

Comment:
Not sure about this one, given that it's stored inside a `PNode` that could harbor a cycle.

Reply:
If a diagnostic stores a tree that references the diagnostic itself then that's a severe bug.

Reply:
Perhaps a note then?

Reply:
I'll add a note.